File: backend/app/schemas/analytics.py
from typing import List, Optional
from pydantic import BaseModel

class KPISummary(BaseModel):
    total_candidates: int
    active_candidates: int
    archived_candidates: int
    total_reviews: int
    average_score: Optional[float] = None
    review_coverage_pct: float

class FunnelStageMetric(BaseModel):
    stage: str
    label: str
    count: int
    percentage: float

class RoleMetric(BaseModel):
    role: str
    candidate_count: int
    average_score: Optional[float] = None
    percentage: float

class CategoryMetric(BaseModel):
    category: str
    average_score: float
    review_count: int

class ScoreDistributionMetric(BaseModel):
    score: int
    count: int
    percentage: float

class SkillMetric(BaseModel):
    skill: str
    count: int

class TopCandidateMetric(BaseModel):
    id: str
    name: str
    role_applied: str
    status: str
    average_score: float
    reviews_count: int

class ReviewerActivityMetric(BaseModel):
    reviewer_id: str
    reviewer_email: str
    reviews_count: int
    average_score_given: Optional[float] = None

class PersonalReviewerStats(BaseModel):
    my_reviews_count: int
    my_candidates_reviewed: int
    my_average_score: Optional[float] = None

class AnalyticsResponse(BaseModel):
    kpis: KPISummary
    funnel: List[FunnelStageMetric]
    roles: List[RoleMetric]
    categories: List[CategoryMetric]
    score_distribution: List[ScoreDistributionMetric]
    top_skills: List[SkillMetric]
    top_candidates: List[TopCandidateMetric]
    reviewer_activity: Optional[List[ReviewerActivityMetric]] = None
    my_stats: Optional[PersonalReviewerStats] = None

File: frontend/src/types/analytics.ts
export interface KPISummary {
  total_candidates: number;
  active_candidates: number;
  archived_candidates: number;
  total_reviews: number;
  average_score: number | null;
  review_coverage_pct: number;
}

export interface FunnelStageMetric {
  stage: string;
  label: string;
  count: number;
  percentage: number;
}

export interface RoleMetric {
  role: string;
  candidate_count: number;
  average_score: number | null;
  percentage: number;
}

export interface CategoryMetric {
  category: string;
  average_score: number;
  review_count: number;
}

export interface ScoreDistributionMetric {
  score: number;
  count: number;
  percentage: number;
}

export interface SkillMetric {
  skill: string;
  count: number;
}

export interface TopCandidateMetric {
  id: string;
  name: string;
  role_applied: string;
  status: string;
  average_score: number;
  reviews_count: number;
}

export interface ReviewerActivityMetric {
  reviewer_id: string;
  reviewer_email: string;
  reviews_count: number;
  average_score_given: number | null;
}

export interface PersonalReviewerStats {
  my_reviews_count: number;
  my_candidates_reviewed: number;
  my_average_score: number | null;
}

export interface AnalyticsResponse {
  kpis: KPISummary;
  funnel: FunnelStageMetric[];
  roles: RoleMetric[];
  categories: CategoryMetric[];
  score_distribution: ScoreDistributionMetric[];
  top_skills: SkillMetric[];
  top_candidates: TopCandidateMetric[];
  reviewer_activity?: ReviewerActivityMetric[] | null;
  my_stats?: PersonalReviewerStats | null;
}
